pygmt/src/shift_origin.py

```diff
@@ -2,20 +2,41 @@
 shift_origin - Shift plot origin in x and/or y directions.
 """
 
+from contextlib import contextmanager
+
 from pygmt.clib import Session
+from pygmt.helpers import build_arg_list
 
 
 def shift_origin(
     self, xshift: float | str | None = None, yshift: float | str | None = None
 ):
     r"""
-    Shift plot origin in x and/or y directions.
+    Shift the plot origin in x and/or y directions.
+
+    The shifts can be permanent or temporary. If used as a standalone method, the shifts
+    are permanent and apply to all subsequent plots. If used as a context manager, the
+    shifts are temporary and only apply to the block of code within the context manager.
+
+    1.  Use as a standalone method to shift the plot origin permanently:
+
+        .. code-block:: python
+
+            fig.shift_origin(xshift=12)
+            ...  # Other plot commands
+
+    2.  Use as a context manager to shift the plot origin temporarily:
+
+        .. code-block:: python
+
+            with fig.shift_origin(...):
+                ...  # Other plot commands
+                ...
```

Comment:
Do we need a second line with dots here? 

```suggestion
        .. code-block:: python

            with fig.shift_origin(...):
                ...  # Other plot commands
```

Reply:
not necessary. I added the two lines to indicate that multiple plotting methods can be called in the code block